src/BenchmarkDotNet/Toolchains/Executor.cs

```diff
@@ -69,7 +69,11 @@ private ExecuteResult Execute(Process process, BenchmarkCase benchmarkCase, Sync
 
             loggerWithDiagnoser.ProcessInput();
 
-            process.WaitForExit(); // should we add timeout here?
+            // Timeout Added. Freeze detected.
+            if (!process.WaitForExit(30000)) 
```

Comment:
30000 ms is 30 s which is definitely too little for majority of the benchmarks we run.

Please revert this change and create a new issue where we can discuss what the default timeout should be. We should also make it configurable